src/components/Ref/Ref.tsx

```diff
@@ -0,0 +1,35 @@
+import * as PropTypes from 'prop-types'
+import * as _ from 'lodash'
+import { Children, Component } from 'react'
+import { findDOMNode } from 'react-dom'
+import { Extendable } from '../../../types/utils'
+
+export interface IRefProps {
+  innerRef?: (ref: HTMLElement) => void
+}
+
+/**
+ * This component exposes a callback prop that always returns the DOM node of both functional and class component
+ * children.
+ */
+export default class Ref extends Component<Extendable<IRefProps>, any> {
+  static propTypes = {
+    /** Primary content. */
+    children: PropTypes.element,
+
+    /**
+     * Called when componentDidMount.
+     *
+     * @param {HTMLElement} node - Referred node.
+     */
+    innerRef: PropTypes.func,
+  }
+
+  componentDidMount() {
+    _.invoke(this.props, 'innerRef', findDOMNode(this))
```

Comment:
There was a condition in SUIR code, let's use it there, too

Reply:
not sure it's necessary since `_.invoke` invokes `innerRef` function only if it's defined; isn't it the same as SUIR?